client/ayon_blender/api/lib.py

```diff
@@ -311,23 +311,38 @@ def get_selected_collections():
     return [id for id in ids if isinstance(id, bpy.types.Collection)]
 
 
-def get_selection(include_collections: bool = False) -> List[bpy.types.Object]:
+def get_selection(include_collections: bool = False)-> List[Union[bpy.types.Object, bpy.types.Collection]]:
     """
     Returns a list of selected objects in the current Blender scene.
 
     Args:
         include_collections (bool, optional): Whether to include selected
         collections in the result. Defaults to False.
+        selected_hierarchies (bool): Whether to include all children
+        of selected objects.
 
     Returns:
-        List[bpy.types.Object]: A list of selected objects.
+        List[Union[bpy.types.Object,
+        bpy.types.Collection]]: Selected objects and optionally collections.
     """
-    selection = [obj for obj in bpy.context.scene.objects if obj.select_get()]
+    selection = set(obj for obj in bpy.context.scene.objects if obj.select_get())
 
     if include_collections:
-        selection.extend(get_selected_collections())
+        selection.update(get_selected_collections())
+    selection.update(get_object_children_recursive(selection))
```

Comment:
Please do not put this inside `get_selection`. This should be a separate function call instead for those who need the hierarchy.

Or if you really prefer it inside `get_selection` put it behind an optional argument `include_object_children_recursive=False`. But a separate function call outside of `get_selection` makes more sense to me.